source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -195,6 +195,7 @@ int DefaultCertValidator::doVerifyCertChain(
 
     if (ret <= 0) {
       stats_.fail_verify_error_.inc();
+      ENVOY_LOG(debug, "{}", Utility::getX509VerificationErrorInfo(store_ctx));
```

Comment:
This is not a review and just my comment. I think we can improve `ssl_socket_text.cc` with this feature to verify not only `fail_verify_error` but checking to debug log with `EXPECT_LOG_CONTAINS` to accomplish more fine-grained tests. Do you think about working on it? If not, I can work for it with another PR.

Reply:
yeah, that'll be more accurate. I think that's reasonable if you see it common to verify the log string in envoy unit test.